src/cryptography/hazmat/backends/openssl/dh.py

```diff
@@ -9,14 +9,26 @@
 from cryptography.hazmat.primitives.asymmetric import dh
 
 
-def _dh_cdata_to_parameters(dh_cdata, backend):
+def _dh_params_dup(dh_cdata, backend):
     lib = backend._lib
     ffi = backend._ffi
 
     param_cdata = lib.DHparams_dup(dh_cdata)
     backend.openssl_assert(param_cdata != ffi.NULL)
     param_cdata = ffi.gc(param_cdata, lib.DH_free)
+    if lib.OPENSSL_VERSION_NUMBER < 0x10002000 or lib.CRYPTOGRAPHY_IS_LIBRESSL:
```

Comment:
This feels like the sort of thing we're eventually going to leak memory on in LibreSSL if and when BNparams_dup copies q. (We would make a new q and set it, then lose the reference to the originally duped `q`). I think the only way to make sure we never leak here even if LibreSSL changes is to do a get0 and check to see if q is null.